### ISSUE TITLE
SC problematic for Closed functionality: SC 2.5.8 Target Size (Minimum)

### ISSUE THREAD

**Issue by pday1** [w3c/wcag2ict#293]
From closed functionality survey:
[survey results](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-target-size-notes/results) 

Closed functionality needs a note for this SC.

Previous content was captured in general issue #44, but it should really be in a separate issue to make tracking easier.

**Comment by pday1** [comment]
Proposed content for SC problematic for Closed functionality section:

[2.5.8 Target Size (Minimum)](#target-size-minimum) - This Success Criterion uses CSS pixels for defining the target size. Closed functionality may not use CSS pixels as a standard measurement, but the definition of 'CSS pixel' still applies as described in [Applying “CSS pixel” to Non-Web Documents and Software]. 

If the system supports a density-independent pixel measurement, it should be used in place of CSS pixels. For software designed to run on specific known hardware, a physical size standard would be more straightforward to apply, as calculations for CSS pixels is dependent on the viewing distance and pixel density of the display.

**Comment by maryjom** [comment]
@patrickhlauke While this is true, some non-web software would be only present on a specific hardware configuration. That product may not have available test tooling to check for target size - especially true for many closed functionality products (self-service transaction machines, printers, etc.) that have a fixed size screen and dedicated software developed specifically for that device. It would be easier for developers/testers of such devices to have a fixed size required measurement to attain. This way they can use calipers as a measuring device to check that targets meet the minimum size. I imagine this problem cannot be easily solved by tweaking the definition of CSS pixels or guidance for the SC.

**Comment by patrickhlauke** [comment]
> use calipers as a measuring device to check that targets meet the minimum size.

sure, but that means you're going to invent a new physical measure to test against in this case, which is not present WCAG itself?

**Comment by maryjom** [comment]
No, WCAG2ICT would not invent the measure. I'd like to simply point out this issue in WCAG2ICT and it would be up to standards organizations to take that information and develop the requirement. It is beyond WCAG2ICT scope to define a specific measure and point out caveats with the direct application of WCAG to all types of non-web documents and software.

**Comment by patrickhlauke** [comment]
ah, in that case, sure. had visions of this document here trying to do that...

**Comment by maryjom** [comment]
We are just "interpreting" WCAG into non-web contexts. It's not always a good direct fit, as one would expect, since WCAG was developed for the web using web-based language and assuming implementations using web-based technology. So the notes and guidance we are providing are to point out where that "direct application" of WCAG causes issues and consternation and provide word substitutions and guidance on how to think about the requirements in non-web contexts.  We can't develop requirements, development or test techniques, or substantively change WCAG requirements. So there's a lot of limitations.

**Comment by patrickhlauke** [comment]
to clarify: i'm aware of the limitations, which is why i was worried ("had visions of" in that sense) that here all of a sudden there was a plan to break out of those limitations and to indeed define a physical size. anyway, all good.

**Comment by samogami** [comment]
@pday1 I just removed the removed the word 'may'.

Note: If the system viewing distance and/or pixel density is unknown then approximating the reference pixel as described in [Applying “CSS pixel” to Non-Web Documents and Software](https://github.com/w3c/wcag2ict/issues/293#applying-css-pixel-to-non-web-documents-and-software) is not possible.

**Comment by patrickhlauke** [comment]
a side note here that in the context of target size, it's likely not so much the *viewing distance* that has relevance, particularly when it comes to touch interfaces - regardless of how close or far I have the device from my eyes, it's what the actual "as rendered on screen" physical dimensions of what the target size shakes out to be that matters.

and yes, the larger point here being: while the idealised measure of what a CSS pixel is does rely, in principle, on viewing distance https://drafts.csswg.org/css-values/#reference-pixel ... this has always been more of a theoretical principle, and browsers/OSs/etc have in the end made their own decisions about what a CSS pixel shakes out to be in real-world terms (and in many cases, OSs etc don't even know, as they don't get the information about the specific physical dimensions or pixel density of an attached display/touchscreen...and users may switch to a different resolution anyway which then defeats the whole idea of it having any kind of real relationship with the real world anyway). in short, it's an idealised density-independent measure. but absent any other possible way to define sizes/dimensions of things, it's the only unit of measure authors can actually influence/use (e.g. even if in CSS you used a supposed physical unit like `mm`, behind the scenes - for on-screen presentation - this always refers back to the CSS reference pixel, and is divorced from its real-world measure)

**Comment by maryjom** [comment]
@patrickhlauke I think the problem space here is the applicability to non-touch devices and the UI of the software applications on them. They may be accessible through some method of pointing device that is not touch and it may not be sophisticated as to have an actual operating system or user agent that that normalizes physical pixels to a density-independent pixel.  This SC is not specific to physical touch but "pointer" which would include mouse pointer, infrared pointer, gaze interaction, etc. that would operate a pointer.

**Comment by olivia-hogan-stark** [comment]
@samogami @pday1 I like the idea of the note addition. Did some word-smithing - feel free to take or leave. 

> Note: If the viewing distance or pixel density of the system is unknown, approximating the reference pixel as described in [Applying “CSS pixel” to Non-Web Documents and Software](https://github.com/w3c/wcag2ict/issues/293#applying-css-pixel-to-non-web-documents-and-software) is not possible. 

Do we need a new Note # to separate out the second part?

**Comment by olivia-hogan-stark** [comment]
@samogami @pday1 I like the idea of the note addition. Do we need a new Note # to separate out the second part?

**Comment by patrickhlauke** [comment]
> This SC is not specific to physical touch but "pointer" which would include mouse pointer, infrared pointer, gaze interaction, etc. that would operate a pointer.

yes i'm aware of that, but again when CSS pixels were chosen as the unit of measure for the SC, the discussion was not centered around "viewing distance" etc. it was only chosen because it's the only reliable unit of measure that authors can choose/control.

in any case, by all means mention that on certain platforms, it's simply not possible to directly "Apply CSS pixel ...", but it's the suggestion here that it's due to lack of information on density/viewing distance that is causing this that doesn't sit right with me, as it does suggest (incorrectly) that the SC chose CSS px with that in mind, when it didn't.

**Comment by pday1** [comment]
Further tweaks incorporating Sam & Olivia's suggestions:

https://github.com/w3c/wcag2ict/issues/293#target-size-minimum - This Success Criterion uses CSS pixels for defining the target size. Closed functionality may not use CSS pixels as a standard measurement, but the definition of 'CSS pixel' still applies as described in [Applying “CSS pixel” to Non-Web Documents and Software](https://github.com/w3c/wcag2ict/issues/293#applying-css-pixel-to-non-web-documents-and-software). If the system supports a density-independent pixel measurement, it should be used in place of CSS pixels.

> Note 1: If the viewing distance or pixel density of the system is unknown, approximating the reference pixel as described in [Applying “CSS pixel” to Non-Web Documents and Software](https://github.com/w3c/wcag2ict/issues/293#applying-css-pixel-to-non-web-documents-and-software) is not possible.

> Note 2: For software designed to run on specific known hardware, a physical size standard would be more straightforward to apply, as calculations for CSS pixels is dependent on the viewing distance and pixel density of the display.

**Comment by maryjom** [comment]
New proposal is in PR #300. [Survey of new draft](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-miscellaneous-changes/) is open until 30 January 2024.

**Comment by maryjom** [comment]
Due to [survey results on 2.5.8](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-miscellaneous-changes/results), the language was slightly adjusted per the [meeting minutes](https://www.w3.org/2024/02/01-wcag2ict-minutes#t03) and will be incorporated into the editor's draft.

**Comment by maryjom** [comment]
The content from PR #300 has been merged, so this can be closed.
